frontend/src/tests/components/Commons/CommonsForm.test.js

```diff
@@ -153,8 +153,8 @@ describe("CommonsForm tests", () => {
 
   it("Check Default Values and correct styles", async () => {
 
-    const curr = new Date();
-    const today = curr.toISOString().substr(0, 10);
+    const curr = new Date(Date.now()-(new Date()).getTimezoneOffset()*60000)
+    const today = curr.toISOString().split('T')[0]
     const currMonth = curr.getMonth() % 12;
     const nextMonth = new Date(curr.getFullYear(), currMonth + 1, curr.getDate()).toISOString().substr(0, 10);
     const DefaultVals = {
```

Comment:
Do you see how this code is exactly the same as the code in the other three places?  Like five conscutive lines of code that just do exactly the same thing?

We should refactor that so that we only have these lines of code in one place.

There's a file called main/utils/dateUtils.js  where you can put utility date functions.

Write a function:

````javascript
const getTodayCurrMonthNextMonth = () =>{
    const curr = new Date(Date.now()-(new Date()).getTimezoneOffset()*60000)
    const today = curr.toISOString().split('T')[0]
    const currMonth = curr.getMonth() % 12;
    const nextMonth = new Date(curr.getFullYear(), currMonth + 1, curr.getDate()).toISOString().substr(0, 10);
    return { today, currMonth, nextMonth };
};
```

Then you can replace each of these sections of code with:
```
const {today, currMonth, nextMonth} = getTodayCurrMonthNextMonth();
```

Reply:
I have made the date getting operation into a function in `dateUtils.js` and added the corresponding test to this new function. I ended up writing the function as `getTodayNextMonth()` as the `currMonth` value is only used internally in the function to calculate `nextMonth` and is not used anywhere else so `currMonth` is dropped from the return values. `currMonth` would have been redundant anyway as `today` should contain the current month. I've also replaced the blocks of code with this function in `CommonsForm.js` and its tests.